vortex-io/src/runtime/current.rs

```diff
@@ -7,51 +7,44 @@ use futures::stream::BoxStream;
 use futures::{Stream, StreamExt};
 use smol::block_on;
 
+pub use crate::runtime::pool::CurrentThreadWorkerPool;
 use crate::runtime::{BlockingRuntime, Executor, Handle};
 
-/// A current thread runtime allows users to explicitly drive Vortex futures from multiple worker
-/// threads that they manage. This is useful in environments where the user already has a thread
-/// pool and wants to integrate Vortex into that pool, for example query engines.
+/// A current thread runtime allows callers to much more explicitly drive Vortex futures than with
+/// a Tokio runtime.
+///
+/// The current thread runtime will do no work unless `block_on` is called. In other words, the
+/// default behavior is single-threaded with code running on the thread that called `block_on`.
+///
+/// It's also possible to clone the runtime onto other threads, each of which can call `block_on`
+/// to drive work on that thread. Each thread shares the same underlying executor with the same
+/// set of tasks, allowing work to be driven in parallel.
+///
+/// For automatic driving of work, a [`CurrentThreadWorkerPool`] can be created from the runtime
+/// by calling [`new_pool`](CurrentThreadRuntime::new_pool). The returned pool can be configured
+/// with the desired number of worker threads that will drive work on behalf of the runtime.
 #[derive(Clone, Default)]
 pub struct CurrentThreadRuntime {
     executor: Arc<smol::Executor<'static>>,
 }
 
-impl BlockingRuntime for CurrentThreadRuntime {
-    type BlockingIterator<'a, R: 'a> = CurrentThreadIterator<'a, R>;
-
-    fn handle(&self) -> Handle {
-        let executor: Arc<dyn Executor> = self.executor.clone();
-        Handle::new(Arc::downgrade(&executor))
-    }
-
-    fn block_on<F, Fut, R>(&self, f: F) -> R
-    where
-        F: FnOnce(Handle) -> Fut,
-        Fut: Future<Output = R>,
-    {
-        block_on(self.executor.run(f(self.handle())))
-    }
-
-    fn block_on_stream<'a, F, S, R>(&self, f: F) -> Self::BlockingIterator<'a, R>
-    where
-        F: FnOnce(Handle) -> S,
-        S: Stream<Item = R> + Send + 'a,
-        R: Send + 'a,
-    {
-        CurrentThreadIterator {
-            executor: self.executor.clone(),
-            stream: f(self.handle()).boxed(),
-        }
-    }
-}
-
 impl CurrentThreadRuntime {
     /// Create a new current thread runtime.
     pub fn new() -> Self {
         Self::default()
     }
 
+    /// Create a new worker pool for driving the runtime in the background.
+    ///
+    /// This pool can be used to offload work from the current thread to a set of worker threads
+    /// that will drive the runtime's executor.
+    ///
+    /// By default, the pool has no worker threads; the caller must set the desired number of
+    /// worker threads using the `set_workers` method on the returned pool.
+    pub fn new_pool(&self) -> CurrentThreadWorkerPool {
```

Comment:
what is the semantic of calling this method twice and having two CurrentThreadWorkerPools pointed at the same CurrentThreadRuntime?

Why isn't managment of workers part of the CurrentThreadRuntime?